src/services/sync-checker.ts

```diff
@@ -48,23 +49,25 @@ export class SyncChecker {
     const syncedNodesKey = `sync-check-${sessionHash}`
     const syncedNodesCached = await this.redis.get(syncedNodesKey)
 
-    if (syncedNodesCached) {
+    const cached = !!syncedNodesCached
```

Comment:
What is this nonsense double negative type conversion

Reply:
welcome to js-land where nothing as expected, this double negative is no error, is just a way of trully converting a value that is "falsey" to a fully boolean, on this case, `syncedNodesCached` could come as null which is a falsey datatype but can throw unexpected errors if treated as a fully boolean, with double exclamation mark we are forcing the null value to become a true boolean value. For more info check [this](https://stackoverflow.com/questions/29312123/how-does-the-double-exclamation-work-in-javascript)

Reply:
This is old stuff @rem1niscence . You can just do 
```suggestion
    const cached = Boolean(syncedNodesCached)
```

That answer is from 2015. Don't blame the language. :)